src/controllers/tripsController.js

```diff
@@ -509,6 +510,28 @@ class tripsController {
       return errorResponse(res, 500, error.message);
     }
   }
+
+  static async tripStatistics(req, res) {
+    try {
+      const {
+        body: {
+          from, until
+        },
+        user: {
+          id, preferedLang
+        }
+      } = req;
+
+      if (new Date(from).getTime() > new Date(until).getTime()) {
+        return errorResponse(res, 422, setLanguage(preferedLang).__('invalidFromDate'));
+      }
```

Comment:
Hi I noticed that you used Joi for date validation 
and i don't get why your validating date again over here in the controllers

Reply:
Hi @blaise, I used Joi to just validate the format of the dates and that they are required, this validation in the controller am just making sure that a user doesn't provide the from date which is in the future because that timeframe would be impossible.

Reply:
yeah I noticed 
I was just saying that joi can handle all the validations